src/Tasks/CultureInfoCache.cs

```diff
@@ -55,7 +57,35 @@ static CultureInfoCache()
         /// <returns>True if the culture is determined to be valid.</returns>
         internal static bool IsValidCultureString(string name)
         {
-            return ValidCultureNames.Contains(name);
+            var isValid = ValidCultureNames.Contains(name);
+            if (isValid)
+                return true;
+
+            var isInvalid = InvalidCultureNames.Contains(name);
+            if (isInvalid)
+                return false;
+
+            CultureInfo culture;
+            try
+            {
+                culture = new CultureInfo(name);
+            }
+            catch (Exception)
+            {
+                InvalidCultureNames.Add(name);
```

Comment:
Since this `catch` is super broad, please comment with an explanation of what it's doing, maybe something like

```suggestion
            catch (Exception)
            {
                // Treat any failure of CultureInfo construction as information that
                // the name is invalid.
                InvalidCultureNames.Add(name);
```

Reply:
Right, I changed to catch `CultureNotFoundException` instead of `Exception` in 19c9e110a6cedb063c6eb7d6555d6d2dad1df022